week1/1.5/Notes/Notes/Controllers/View Controllers/NoteTableViewController.swift

```diff
@@ -0,0 +1,136 @@
+        super.viewDidLoad()
+        NoteController.shared.loadFromPersistentStore()
+        searchController.searchResultsUpdater = self
+        searchController.obscuresBackgroundDuringPresentation = false
+        searchController.searchBar.placeholder = "Search notes"
+        navigationItem.searchController = searchController
+        navigationItem.hidesSearchBarWhenScrolling = false
+        definesPresentationContext = true
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        tableView.reloadData()
+    }
+    
+    // MARK: - Table view data source
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if isFiltering() {
+            return filteredNotes.count
+        }
+        return NoteController.shared.notes.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "noteCell", for: indexPath)
+        let note = findCorrectNote(withIndex: indexPath)
+        cell.textLabel?.text = note.bodyText
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCellEditingStyle, forRowAt indexPath: IndexPath) {
+        if isFiltering() {
+            if editingStyle == .delete {
+                let note = findCorrectNote(withIndex: indexPath)
+                NoteController.shared.deleteNote(note)
+                tableView.reloadData()
+                tableView.deleteRows(at: [indexPath], with: .fade)
+            }
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, canMoveRowAt indexPath: IndexPath) -> Bool {
+        return true
+    }
+    
+    override func tableView(_ tableView: UITableView, moveRowAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath) {
+        let note: Note = NoteController.shared.notes[sourceIndexPath.row]
+        NoteController.shared.deleteNote(note)
+        NoteController.shared.addNote(note)
+    }
+    
+    // MARK: - Private instance methods
+    func searchBarIsEmpty() -> Bool {
+        // Returns true if the text is empty or nil
+        return searchController.searchBar.text?.isEmpty ?? true
+    }
+    
+    func filterContentForSearchText(_ searchText: String, scope: String = "All") {
+        filteredNotes = NoteController.shared.notes.filter({( note : Note) -> Bool in
+            return note.bodyText.lowercased().contains(searchText.lowercased())
+            
+        })
+        tableView.reloadData()
+    }
+    
+    func isFiltering() -> Bool {
+        if tableView.isEditing {
+            tableView.setEditing(false, animated: true)
+            editBarButton.title = "Edit";
+            editBarButton.style = UIBarButtonItemStyle.plain;
+        }
+        return searchController.isActive && !searchBarIsEmpty()
```

Comment:
This isn't too bad, but ugly combinations of multiple different `not`s is ugly. The emptiness is a negative condition. The `!` is a negative condition. This ends up being "active and not not occupied."